core/shared/src/main/scala/org/typelevel/idna4s/core/bootstring/Damp.scala

```diff
@@ -21,44 +21,102 @@
 
 package org.typelevel.idna4s.core.bootstring
 
-import cats._
+import cats.Show
+import cats.kernel._
 import cats.syntax.all._
 
-sealed abstract class Damp extends Product with Serializable {
-  def value: Int
-
+/**
+ * Newtype for RFC 3492 Damp value.
+ *
+ * A damp is a special value used in the bias adaptation function of Bootstring when scaling the
+ * very first delta. It compensates for the fact that the second delta is usually much smaller
+ * the first.
+ *
+ * @see
+ *   [[https://datatracker.ietf.org/doc/html/rfc3492#section-3.4 Bias Adaptation]]
+ */
+final class Damp private (val value: Int) extends AnyVal {
   final override def toString: String = s"Damp(value = ${value})"
 }
 
 object Damp {
-  final private[this] case class DampImpl(override val value: Int) extends Damp
+  private def apply(value: Int): Damp =
+    new Damp(value)
 
-  val PunycodeDamp: Damp = unsafeFromInt(700)
+  val MinValue: Damp = Damp(2)
+  val MaxValue: Damp = Damp(Int.MaxValue)
 
-  def fromInt(value: Int): Either[String, Damp] =
+  /**
+   * The damp used by the Punycode Bootstring invocation.
+   */
+  final val PunycodeDamp: Damp = new Damp(700)
```

Comment:
can this use the literal syntax?

```suggestion
  final val PunycodeDamp: Damp = damp"700"
```

Reply:
No, because it's the same compilation unit.

Reply:
ahh, my bad.